spec/ruby-push-notifications/wns/wns_notification_spec.rb

```diff
@@ -0,0 +1,68 @@
+module RubyPushNotifications
+  module WNS
+    describe WNSNotification do
+
+      let(:device_urls) { %w(a b c) }
+      let(:raw_notification) { build :wns_notification }
+      let(:toast_data) { { title: 'Title', message: 'Hello WNS World!', type: :toast } }
+      let(:toast_notification) { build :wns_notification, device_urls: device_urls, data: toast_data }
+      let(:tile_data) { { message: 'Hello WNS World!', image: 'http://image.com/image.jpg', type: :tile } }
+      let(:tile_notification) { build :wns_notification, device_urls: device_urls, data: tile_data }
+      let(:badge_data) { { value: 10, type: :badge } }
+      let(:badge_notification) { build :wns_notification, device_urls: device_urls, data: badge_data }
+
+      it 'builds the right wns raw xml' do
```

Comment:
I'd like to have a bit more complicated test for this case to have the recursive `build_hash` covered

Reply:
added